main.go

```diff
@@ -88,6 +88,9 @@ func findIDEDistrubutionLogsPath(output string) (string, error) {
 func macCodeSignGroup(archive xcarchive.MacosArchive, installedCertificates []certificateutil.CertificateInfoModel,
 	installedInstallerCertificates []certificateutil.CertificateInfoModel, installedProfiles []profileutil.ProvisioningProfileInfoModel,
 	exportMethod exportoptions.Method, cfg config) (*export.MacCodeSignGroup, error) {
+	if archive.Application.ProvisioningProfile == nil {
+		return nil, fmt.Errorf("precondition false, provisioning profile expected in the archive")
```

Comment:
Why this check is needed here if it was not present so far? This is fixing a bug?

Reply:
No, added to be safe as this function only works with provisioning profiles. If no provisioning profile is in the archive, it is not called, so it could make sense to document this as a precondition. I am open to suggestions on how this should look like. It could be just a comment too.

Reply:
That makes sense. I think that assert is good like that. I just didn't understand why it appeared. I think it is good this way